tracer/src/Datadog.Trace.Coverage.collector/CoverageCollector.cs

```diff
@@ -0,0 +1,275 @@
+// <copyright file="CoverageCollector.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using System.Xml;
+using Datadog.Trace.Ci.Configuration;
+using Datadog.Trace.Ci.Coverage.Exceptions;
+using Datadog.Trace.ClrProfiler;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+using Formatting = Datadog.Trace.Vendors.Newtonsoft.Json.Formatting;
+
+// Disable SA1300 Warning due the `collector` lowercase in the namespace
+#pragma warning disable SA1300
+namespace Datadog.Trace.Coverage.collector
+{
+    /// <summary>
+    /// Datadog coverage collector
+    /// </summary>
+    [DataCollectorTypeUri("datacollector://Datadog/CoverageCollector/1.0")]
+    [DataCollectorFriendlyName("DatadogCoverage")]
+    public class CoverageCollector : DataCollector
+    {
+        private DataCollectorLogger? _logger;
+        private DataCollectionEvents? _events;
+        private CIVisibilitySettings? _ciVisibilitySettings;
+        private DateTime _dateTime = DateTime.Now;
+        private string? _tracerHome;
+
+        private static void Copy(string sourceDirectory, string targetDirectory)
+        {
+            var diSource = new DirectoryInfo(sourceDirectory);
+            var diTarget = new DirectoryInfo(targetDirectory);
+
+            CopyAll(diSource, diTarget);
+        }
+
+        private static void CopyAll(DirectoryInfo source, DirectoryInfo target)
+        {
+            var files = source.GetFiles();
+            var subFolders = source.GetDirectories();
+
+            Directory.CreateDirectory(target.FullName);
+
+            // Copy each file into the new directory.
+            foreach (var fi in files)
+            {
+                fi.CopyTo(Path.Combine(target.FullName, fi.Name), true);
+            }
+
+            // Copy each subdirectory using recursion.
+            foreach (var diSourceSubDir in subFolders)
+            {
+                var nextTargetSubDir = target.CreateSubdirectory(diSourceSubDir.Name);
+                CopyAll(diSourceSubDir, nextTargetSubDir);
+            }
+        }
+
+        /// <inheritdoc />
+        public override void Initialize(XmlElement configurationElement, DataCollectionEvents events, DataCollectionSink dataSink, DataCollectionLogger logger, DataCollectionEnvironmentContext environmentContext)
+        {
+            _events = events;
+            _logger = new DataCollectorLogger(logger, environmentContext.SessionDataCollectionContext);
+            Console.SetOut(_logger.GetTextWriter());
```

Comment:
How come `Console.Out` is getting redirected? Was this written to aid development or is this needed for functionality?

Reply:
you're right it was for debugging, removed in [d0d8115](https://github.com/DataDog/dd-trace-dotnet/pull/2739/commits/d0d8115612e93d65e6a09402f69f3443e1a35bfb)